python/pyspark/pandas/correlation.py

```diff
@@ -219,6 +219,43 @@ def compute(sdf: SparkDataFrame, groupKeys: List[str], method: str) -> SparkData
             F.col(CORRELATION_VALUE_2_COLUMN) == F.col(CORRELATION_VALUE_Y_COLUMN)
         )
 
+        spark_session = SparkSession.getActiveSession()
+        if is_ansi_mode_enabled(spark_session):
+            corr_expr = F.try_divide(
```

Comment:
Can we just use this expression even when ANSI mode is disabled? checking the configuration is actually expensive..